framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -1321,6 +1330,115 @@ MeshDiagnosticsGenerator::checkLocalJacobians(const std::unique_ptr<MeshBase> &
                  num_negative_side_qp_jacobians);
 }
 
+void
+MeshDiagnosticsGenerator::checkNonMatchingEdges(const std::unique_ptr<MeshBase> & mesh) const
+{
+  /*Algorithm Overview
+    1)Prechecks
+      a)This algorithn only works for 3D so check for that first
+    2)Loop
+      a)Loop through every element
+      b)For each element get the edges associated with it
+      c)For each edge check overlap with any edges of nearby elements
+      d)Have check to make sure the same pair of edges are not being tested twice for overlap
+    3)Overlap check
+      a)Shortest line that connects both lines is perpendicular to both lines
+      b)A good overview of the math for finding intersecting lines can be found
+    here->paulbourke.net/geometry/pointlineplane/
+  */
+  if (mesh->mesh_dimension() != 3)
+    mooseError("The edge intersection algorithm only works with 3D meshes");
+  if (!mesh->is_serial())
+    mooseError("Only serialized/replicated meshes are supported");
+  unsigned int num_intersecting_edges = 0;
+  std::set<Node> overlapping_edges_nodes;
+  for (const auto elem : mesh->active_element_ptr_range())
+  {
+    std::vector<std::unique_ptr<Elem>> elem_edges(elem->n_edges());
+    for (auto i : elem->edge_index_range())
+      elem_edges[i] = elem->build_edge_ptr(i);
+    for (const auto other_elem : mesh->active_element_ptr_range())
+    {
+      // If they're the same element then there's no need to check for overlap
+      if (elem == other_elem)
+        continue;
+
+      // Get bounding boxes for both elements. If the bounding boxes don't intersect then no edges
+      // will either
+      const auto boundingBox1 = elem->loose_bounding_box();
+      const auto boundingBox2 = other_elem->loose_bounding_box();
+      if (!(boundingBox1.intersects(boundingBox2)))
+        continue;
+
+      std::vector<std::unique_ptr<Elem>> other_edges(other_elem->n_edges());
+      for (auto j : other_elem->edge_index_range())
+        other_edges[j] = other_elem->build_edge_ptr(j);
+      for (auto & edge : elem_edges)
+      {
+        for (auto & other_edge : other_edges)
+        {
+          // Get nodes from edges
+          const Node * n1 = edge->get_nodes()[0];
+          const Node * n2 = edge->get_nodes()[1];
+          const Node * n3 = other_edge->get_nodes()[0];
+          const Node * n4 = other_edge->get_nodes()[1];
+
+          // Check if the edges have already been added to our check_edges list
+          if (overlapping_edges_nodes.count(*n1) && overlapping_edges_nodes.count(*n2) &&
+              overlapping_edges_nodes.count(*n3) && overlapping_edges_nodes.count(*n4))
+          {
+            continue;
+          }
+
+          // Check element/edge type
+          if (edge->type() != EDGE2)
+          {
+            std::string element_message = "Edge of type " + Utility::enum_to_string(edge->type()) +
```

Comment:
you're missing the include "enum_to_string.h" (or whatever it's really called) and that's why the non-unity build is failing

Reply:
Thanks! I was trying to figure out why that was failing. I'll fix that right now.